Sources/NIO/EventLoop.swift

```diff
@@ -273,13 +273,47 @@ public protocol EventLoop: EventLoopGroup {
     /// Contrary to `makeSucceededFuture`, `makeSucceededVoidFuture` is a customization point for `EventLoop`s which
     /// allows `EventLoop`s to cache a pre-succeded `Void` future to prevent superfluous allocations.
     func makeSucceededVoidFuture() -> EventLoopFuture<Void>
+
+    /// Debug hook: track a promise creation and its location.
```

Comment:
Should we just extract these two functions into another internal-underscored `protocol` and have `EventLoop` implementations optionally support it? It'd be less visible to someone holding an `EventLoop` and we don't have to add the default implementations.

Reply:
I think I'd rather not do it: things that can be safely defaulted without any damage should be just fine. Users are highly unlikely to call this method directly, and even if they did it's not like anything bad would happen. Underscored methods don't appear in the autocomplete unless you deliberately attempt to find them, so they don't impose any particular cognitive overhead on users. They also save us from needing runtime type checks to validate a given Event Loop's conformance.

Reply:
True, it just feels slightly off to me to add internal debug hooks here. It's not a big deal though.

Reply:
I mean, they're not entirely internal: we're happy for third-party implementers of event loops to implement them if they want to. For example, NIOTS will.